components/dash-core-components/src/components/Input.react.js

```diff
@@ -66,13 +66,15 @@ export default class Input extends PureComponent {
                 data-dash-is-loading={
                     (loading_state && loading_state.is_loading) || undefined
                 }
+                className={['dash-input', this.props.className].join(' ')}
```

Comment:
This still puts an empty space at the end if undefined. The most proper way I can see to do it, would be to instantiate an array with `['dash-input']` and add `className` if not undefined then join the array or filter is probably enough for such a case with only 2 elements.